RevenueCatUI/CustomerCenter/ViewModels/BaseManageSubscriptionViewModel/BaseManageSubscriptionViewModel.swift

```diff
@@ -0,0 +1,337 @@
+        self.inAppBrowserURL = nil
+    }
+
+    func openManagementURL(for productId: String) {
+        guard let purchaseInformation,
+              purchaseInformation.productIdentifier == productId,
+        let url = purchaseInformation.managePurchaseURL else {
+            return
+        }
+
+        self.inAppBrowserURL = IdentifiableURL(url: url)
+    }
+
+#endif
+
+}
+
+// MARK: - Promotional Offer Sheet Dismissal Handling
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+extension BaseManageSubscriptionViewModel {
+
+    /// Function responsible for handling the user's action on the PromotionalOfferView
+    func handleDismissPromotionalOfferView(_ userAction: PromotionalOfferViewAction) async {
+        switch userAction {
+        case .successfullyRedeemedPromotionalOffer:
+            self.actionWrapper.handleAction(.promotionalOfferSuccess)
+        case .declinePromotionalOffer, .promotionalCodeRedemptionFailed:
+            break
+        }
+
+        // Clear the promotional offer data to dismiss the sheet
+        self.promotionalOfferData = nil
+
+        if userAction.shouldTerminateCurrentPathFlow {
+            self.loadingPath = nil
+        } else {
+            if let loadingPath = loadingPath {
+                await self.onPathSelected(path: loadingPath)
+                self.loadingPath = nil
+            }
+        }
+    }
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+private extension BaseManageSubscriptionViewModel {
+
+#if os(iOS) || targetEnvironment(macCatalyst)
+    // swiftlint:disable:next cyclomatic_complexity
+    private func onPathSelected(path: CustomerCenterConfigData.HelpPath) async {
+        switch path.type {
+        case .missingPurchase:
+            self.showRestoreAlert = true
+
+        case .refundRequest:
+            guard let purchaseInformation = self.purchaseInformation else { return }
+            let productId = purchaseInformation.productIdentifier
+            do {
+                self.actionWrapper.handleAction(.refundRequestStarted(productId))
+
+                let status = try await self.purchasesProvider.beginRefundRequest(forProduct: productId)
+                self.refundRequestStatus = status
+                self.actionWrapper.handleAction(.refundRequestCompleted(productId, status))
+            } catch {
+                self.refundRequestStatus = .error
+                self.actionWrapper.handleAction(.refundRequestCompleted(productId, .error))
+            }
+
+        case .cancel where purchaseInformation?.store != .appStore:
+            if let url = purchaseInformation?.managePurchaseURL {
+                self.inAppBrowserURL = IdentifiableURL(url: url)
+            } else {
+                // WIP: shor error alert?
```

Comment:
When would this happen? I think we should not show the cancel button if there's no `managePurchaseURL` if `!= .appStore`

Reply:
I know, so maybe an error log is enough here?

Reply:
Yeah, that's why I thought an error alert would make sense (in case it happens??) 
Or maybe a log

Reply:
Just added a test to filter the cancel if there's no URL 🚀